service/forward_service.cc

```diff
@@ -200,23 +213,27 @@ future<query::forward_result> forward_service::dispatch(query::forward_request r
         vnodes_per_addr[endpoint_addr].push_back(*vnode);
     }
 
+    tracing::trace(tr_state, "Dispatching forward_request to {} endpoints", vnodes_per_addr.size());
+
+    std::optional<tracing::trace_info> tr_info = tracing::make_trace_info(tr_state);
     std::optional<query::forward_result> result;
     // Forward request to each endpoint and merge results.
     co_await parallel_for_each(vnodes_per_addr.begin(), vnodes_per_addr.end(),
-        [this, &req, &result] (auto vnodes_with_addr) -> future<> {
+        [this, &req, &result, &tr_state, &tr_info] (auto vnodes_with_addr) -> future<> {
             auto& addr = vnodes_with_addr.first;
             auto& partition_range = vnodes_with_addr.second;
             auto req_with_modified_pr = req;
             req_with_modified_pr.pr = partition_range;
 
+            tracing::trace(tr_state, "Sending forward_request to {}", addr);
             flogger.debug("dispatching forward_request={} to address={}", req_with_modified_pr, addr);
 
             query::forward_result partial_result = co_await [&] {
                 if (utils::fb_utilities::is_me(addr.addr)) {
-                    return execute(req_with_modified_pr);
+                    return execute(req_with_modified_pr, tr_info);
                 } else {
                     return ser::forward_request_rpc_verbs::send_forward_request(
-                        &_messaging, addr, req_with_modified_pr
+                        &_messaging, addr, req_with_modified_pr, tr_info
                     );
                 }
```

Comment:
We should have a tracepoint for the result, and one at the completion of parallel_for_each().

Reply:
(the completion tracepoints are much more interesting, since they have variability in time)